ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFDateDiff.java

```diff
@@ -110,17 +111,18 @@ private Date convertToDate(PrimitiveCategory inputType, Converter converter, Def
     switch (inputType) {
     case STRING:
     case VARCHAR:
-    case CHAR:
+    case CHAR: {
```

Comment:
Why the barces? We don't use them anywhere else, no need for them, in a switch the code after a case is a block by default.

Reply:
`Timestamp ts` is declared twice: in CHAR and TIMESTAMP.  Needs the block or won't compile for having two variables with same name,